File: alwin-middleware-grapescode/alwin-core-api/src/main/java/com/codersteam/alwin/core/api/service/issue/PersonService.java
package com.codersteam.alwin.core.api.service.issue;

import com.codersteam.alwin.core.api.model.customer.PersonDto;

import javax.ejb.Local;
import java.util.List;

/**
 * Serwis do obsługi osób uprawnionych firmy
 *
 * @author Piotr Naroznik
 */
@Local
public interface PersonService {

    /**
     * Oznaczanie/usuwanie oznacznia osoby jak osoba do kontaktu
     *
     * @param personId      - identyfikator osoby
     * @param companyId     - identyfikator firmy
     * @param contactPerson - czy oznaczyć jako osobę do kontaktu true/false
     */
    void setContactPerson(Long personId, Long companyId, boolean contactPerson);

    /**
     * Dodanie osoby uprawnionej firmy
     *
     * @param companyId - identyfikator firmy
     * @param personDto - dane osoby uprawionej
     */
    void createPerson(Long companyId, PersonDto personDto);


    /**
     * Pobieranie osób uprawnionych firmy
     *
     * @param companyId - identyfikator firmy
     * @return osoby uprawnione
     */
    List<PersonDto> getCompanyPersonsByCompanyId(long companyId);
}

File: alwin-web-grapescode/src/app/person/person.service.ts
import {Injectable} from '@angular/core';
import {AlwinHttpService} from '../shared/authentication/alwin-http.service';
import {ContactPerson} from '../customer/shared/contact-person';
import {Person} from '../customer/shared/person';
import {KeyLabel} from '../shared/key-label';
import {Observable} from 'rxjs';

/**
 * Serwisu dostępu do usług REST dla osób uprawnionych
 */
@Injectable()
export class PersonService {

  constructor(private http: AlwinHttpService) {
  }

  /**
   * Oznacznie lub odznaczenie osoby uprawnionej firmy jako osoba do kontaktu
   * @param {number} personId - identyfikator osoby
   * @param {number} companyId - identyfikator firmy
   * @param {boolean} isContactPerson - czy osoba do kontaktu
   * @returns {any} odpowiedź z poprawnym kodem http aktualizacji 202
   */
  updateContactPerson(personId: number, companyId: number, isContactPerson: boolean): any {
    return this.http.patch(`persons/${personId}/company/${companyId}`, new ContactPerson(isContactPerson));
  }

  /**
   * Dodanie osoby uprawnionej firmy
   * @param {number} companyId - identyfikator firmy
   * @param {Person} person - dane osoby
   * @returns {any} odpowiedź z poprawnym kodem http utworzenia 201 lub błędy walidacji
   */
  createPerson(companyId: number, person: Person): any {
    return this.http.postObserve(`persons/company/${companyId}`, person);
  }

  /**
   * Pobieranie statusów cywilnych
   * @returns {Observable<KeyLabel[]>} statusy cywilne
   */
  getMaritalStatuses(): Observable<KeyLabel[]> {
    return this.http.get<KeyLabel[]>('persons/maritalStatuses');
  }

  /**
   * Pobieranie typów dokumentu tożsamości
   * @returns {Observable<KeyLabel[]>} typy dokumentu tożsamości
   */
  getDocumentTypes(): Observable<KeyLabel[]> {
    return this.http.get<KeyLabel[]>('persons/documentTypes');
  }

  /**
   * Pobieranie osób uprawnionych dla firmy
   * @param {number} companyId - identyfikator firmy
   * @returns {Observable<KeyLabel[]>} osoby uprawione
   */
  getCompanyPersons(companyId: number): Observable<Person[]> {
    return this.http.get<Person[]>(`persons/company/${companyId}`);
  }
}
